include/swift/AST/Decl.h

```diff
@@ -3787,22 +3783,62 @@ class ClassDecl final : public NominalTypeDecl {
   }
 
   Optional<bool> getCachedInheritsSuperclassInitializers() const {
-    if (Bits.ClassDecl.ComputedInheritsSuperclassInits)
-      return Bits.ClassDecl.InheritsSuperclassInits;
+    bool hasAttr =
+        getAttrs().hasAttribute<InheritsConvenienceInitializersAttr>();
+    if (hasAttr || Bits.ClassDecl.ComputedInheritsSuperclassInits)
+      return hasAttr;
+
+    return None;
+  }
+
+  Optional<bool> getCachedHasMissingDesignatedInitializers() const {
+    if (!Bits.ClassDecl.ComputedHasMissingDesignatedInitializers) {
+      // Force loading all the members, which will add this attribute if any of
+      // members are determined to be missing while loading.
+      auto mutableThis = const_cast<ClassDecl *>(this);
+      (void)mutableThis->lookupDirect(DeclBaseName::createConstructor());
+    }
+
+    bool hasAttr = getAttrs().hasAttribute<HasMissingDesignatedInitsAttr>();
```

Comment:
Checking for the attribute should be part of the request's evaluate() method -- it can short-circuit the actual computation there.